### ISSUE TITLE
Reverse Proxy Domain does not work on Synology NAS

### ISSUE THREAD

**Issue by Jainic** [Requarks/wiki-v1#165]
### Problem: When I type my domain in to my browser, the connection hangs. I get this error : ERR_CONNECTION_TIMED_OUT

I am running the **Wiki.js V1** and **mongodb** in a container using docker. 
They are being hosted on my Synology NAS at home (unfortunately I can't test this at home at the moment).
Both containers seem to be working: 

Log of Wiki:
![image](https://user-images.githubusercontent.com/47646390/52821345-88295980-307c-11e9-9df7-fdf8bf8cb918.png)

Log of DB:
![image](https://user-images.githubusercontent.com/47646390/52821333-7e9ff180-307c-11e9-8f9d-b21fce584e26.png)

This is the Reverse Proxy Rules:
![image](https://user-images.githubusercontent.com/47646390/52820944-6d0a1a00-307b-11e9-8652-c1eca84aa986.png)

For your reference, here is the docker-compose.yml file:

![image](https://user-images.githubusercontent.com/47646390/52821469-dcccd480-307c-11e9-977f-eefff31c1832.png)

And here is the config.yml:

![image](https://user-images.githubusercontent.com/47646390/52821532-0b4aaf80-307d-11e9-9480-8afb17a0ba18.png)

### Where did I go wrong?  @NGPixel Please help when you get a chance thank you!

**Comment by NGPixel** [comment]
Try running mongoDB as a separate container. Then run wiki.js. In some instances, the mongo container isn't ready to serve requests by the time wiki.js is booting.

**Comment by Jainic** [comment]
Hi, thank you for replying. I'm puling my hair out at this point.
MongoDB is running in a separate container from the start.. It's really strange, now I see the favicon showing up on the corner. And an 502 error.

![image](https://user-images.githubusercontent.com/47646390/52893281-b97f5380-3168-11e9-8097-69ad430de508.png)

![image](https://user-images.githubusercontent.com/47646390/52893733-c783a300-316d-11e9-8b68-2edd35ef011a.png)

I'm new to this porting rule so I'm not sure if I understand this correctly. Please help me check my logic.
1. Browser  - I type wiki.mydomain.ca:8888
2. Router  - Router port 8888 is port forwarded to local port 8888 
3. Synology Reverse Proxy (built in NGINX) - Reverse proxy sends wiki.mydomain.ca:8888 to container port 3003
4. Docker-compose.yml - Wikijs port is set to 3000:3003 (3000 is host, 3003 is container)
5. Config.yml - Main server port is 3000.

**Comment by Jainic** [comment]
I finally got it working. It was an issue with reverse proxy and firewall.
I hope no one spends as much time as I did so I'll post my solution in case anyone runs into the same troubles.

config.yml:

![image](https://user-images.githubusercontent.com/47646390/53037221-a067f280-3447-11e9-81d4-79be2047ee50.png)
Note: Ensure that the host is the public domain you will enter to visit the website. This must match otherwise your gonna get an ugly wiki without any CSS.

docker-compose.yml:
![image](https://user-images.githubusercontent.com/47646390/53037237-acec4b00-3447-11e9-8147-a8ed42c13fdc.png)

Reverse Proxy (inside Synology NAS):
1. Go to Control Panel
2. Application Portal
3. Reverse Proxy Rules
Note: Destination should be **HTTP** not HTTPS!! 
![image](https://user-images.githubusercontent.com/47646390/53037359-ffc60280-3447-11e9-8e4a-3d45bb4d0a52.png)

Disable firewall (inside Synology NAS):
1. Control Panel 
2. Security
3. Uncheck 'Enable firewall'
Note: It is good practice to enable firewall. Find out which ports you want to keep and add it to the rules (Edit Rules). Then disable all unused ports for security measure.

After 7 days, I have finally made it work. Now I'm going to try installing beta and see which I like better :3
